plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -225,7 +225,7 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
             return Optional.of(new DateToVarcharCoercer(toVarcharType));
         }
         if (fromType == DOUBLE && toType instanceof VarcharType toVarcharType) {
-            return Optional.of(new DoubleToVarcharCoercer(toVarcharType, coercionContext.treatNaNAsNull()));
+            return Optional.of(new DoubleToVarcharCoercer(toVarcharType, coercionContext.isOrcFile()));
```

Comment:
Please rename also the constructor parameter in `DoubleToVarcharCoercer`  and also `treatNaNAsNull` occurences in `TestDoubleToVarcharCoercions`